src/lib/rest/rest.cpp

```diff
@@ -1410,7 +1410,22 @@ static int connectionTreat
   }
   else
   {
-    serveFunction(ciP);
+    if (metricsMgr.isOn())
+    {
+      struct timeval  start;
+      struct timeval  end;
+
+      gettimeofday(&start, NULL);
```

Comment:
Looking after `c02e15a` it seems that the "start time" is in the righ place. However, not sure about the "end time". The connectionTreat() function has many exit points (consisting in restReply + return, e.g. https://github.com/telefonicaid/fiware-orion/blob/b143fe974eaa58003c7488fc3197999eca9a9289/src/lib/rest/rest.cpp#L1385) before the point in which this "end time" is done. There are also two "if branches" apart from the one in which the "end time" is located (here https://github.com/telefonicaid/fiware-orion/blob/b143fe974eaa58003c7488fc3197999eca9a9289/src/lib/rest/rest.cpp#L1418 and here https://github.com/telefonicaid/fiware-orion/blob/b143fe974eaa58003c7488fc3197999eca9a9289/src/lib/rest/rest.cpp#L1423).

(Maybe a function which so many exit points as connectionTreat() if difficult to manage... I have that "feeling" when I refere to #940. Now I see that #940 refers to a different kind of refactor, but I hope you get my point. Having say that, I'm not fully sure that a refactor help... I only have the "feeling" that cleaning things could make things easier).